_posts/en/newsletters/2023-09-13-newsletter.md

```diff
@@ -0,0 +1,186 @@
+can be thought of as implementing a deeper level of serialization and
+deserialization. These classes do not do the actual sending and receiving.
+
+The PR derives two concrete classes from the `Transport` class,
+`V1Transport` (what we have today) and `V2Transport` (encrypted on the
+wire). This PR is part of the
+[BIP324][topic v2 p2p transport]
+_Version 2 P2P Encrypted Transport Protocol_ [project][v2 p2p tracking pr].
+
+{% include functions/details-list.md
+  q0="What is the distinction between [*net*][net] and
+      [*net_processing*][net_processing]?"
+  a0="*Net* is at the bottom of the networking stack and handles
+       low-level communication between peers, while *net_processing*
+       builds on top of the *net* layer and handles the processing
+       and validation of messages from *net* layer."
+  a0link="https://bitcoincore.reviews/28165#l-22"
+
+  q1="More concretely, name examples of classes or functions that
+      we'd associate with *net_processing*, and, in contrast,
+      with *net*?"
+  a1="*net_processing*: `PeerManager`, `ProcessMessage`.
+      *net*: `CNode`, `ReceiveMsgBytes`, `CConnMan`."
+  a1link="https://bitcoincore.reviews/28165#l-25"
+
+  q2="Does BIP324 require changes to the *net* layer, the
+      *net_processing* layer, or both? Does it affect policy
+      or consensus?"
+  a2="These changes are only at the *net* layer; they don't affect consensus."
+  a2link="https://bitcoincore.reviews/28165#l-37"
+
+  q3="What are examples of implementation bugs that could result in
+      this PR being an (accidental) consensus change?"
+  a3="A bug that restricts the maximum message size less to than
+      4MB, resulting in a blockweight soft fork; a bug in block
+      deserialization that causes the node to reject a consensus-valid
+      block, also a soft fork."
+  a3link="https://bitcoincore.reviews/28165#l-45"
+
+  q4="`CNetMsgMaker` and `Transport` both “serialize” messages.
+      What is the difference in what they do?"
+  a4="`CNetMsgMaker` performs the serialization of data structures
+      into bytes; `Transport` receives these bytes, adds
+      (serializes) the header, and actually sends it."
+  a4link="https://bitcoincore.reviews/28165#l-60"
+
+  q5="In the process of turning an application object like a
+      `CTransactionRef` (transaction) into bytes / network packets, what
+      happens? Which data structures does it turn into in the process?"
+  a5="`msgMaker.Make()` serializes the `CTransactionRef` message by
+      calling `SerializeTransaction()`, then `PushMessage()` puts the
+      serialized message into the `vSendMsg` queue, then `SocketSendData()`
+      adds a header/checksum (after the changes from this PR) and asks the
+      transport for the next packet to send, and finally calls `m_sock->Send()`."
+  a5link="https://bitcoincore.reviews/28165#l-83"
+
+  q6="How many bytes are sent over the wire for the `sendtxrcncl`message
```

Comment:
Nit: can we mention what `sendtxrcncl` message does?  I'm parsing it as SendTransactionCancel, but I doubt that's correct.

Reply:
Yeah, it seems that it is “received from a peer demonstrating readiness to announce transactions via reconciliations.”